frontend/src/lib/components/metadata/cells/SliceFinderCell.svelte

```diff
@@ -29,28 +30,24 @@
 	function addSlice() {
 		slice.sliceName = newSliceName;
 		if ($project !== undefined) {
-			ZenoService.addSlice($project.uuid, slice).then(() => {
-				ZenoService.getSlices($project ? $project.uuid : '').then((fetchedSlices) => {
-					slices.set(fetchedSlices);
-					$model !== undefined && updateModelDependentSlices('model A', $model, $slices);
-					$comparisonModel !== undefined &&
-						updateModelDependentSlices('model B', $comparisonModel, $slices);
-					showSliceName = false;
-					created = true;
-				});
+			ZenoService.addSlice($project.uuid, slice).then((res) => {
+				invalidateAll();
+				slices.update((s) => [...s, { ...slice, id: res }]);
+				$model !== undefined && updateModelDependentSlices('model A', $model, $slices);
+				$comparisonModel !== undefined &&
+					updateModelDependentSlices('model B', $comparisonModel, $slices);
+				showSliceName = false;
+				created = true;
 			});
 		}
 	}
 
 	/** Remove a generated slice from the slice drawer **/
 	function removeSlice() {
 		ZenoService.deleteSlice(slice).then(() => {
-			if ($project) {
-				ZenoService.getSlices($project.uuid).then((fetchedSlices) => {
-					slices.set(fetchedSlices);
-					created = false;
-				});
-			}
+			invalidateAll();
+			slices.update((s) => s.filter((sli) => sli.sliceName !== slice.sliceName));
```

Comment:
Should not do by name imo. Can do by id, no?

Reply:
hehe good catch, copilot messin me up.